nylas/client/client.py

```diff
@@ -89,30 +91,54 @@ def __init__(
         )
         self.session.headers = {
             "X-Nylas-API-Wrapper": "python",
-            "X-Nylas-Client-Id": self.app_id,
+            "X-Nylas-Client-Id": self.client_id,
             "User-Agent": version_header,
         }
         self._access_token = None
         self.access_token = access_token
         self.auth_token = None
 
         # Requests to the /a/ namespace don't use an auth token but
-        # the app_secret. Set up a specific session for this.
+        # the client_secret. Set up a specific session for this.
         self.admin_session = requests.Session()
 
-        if app_secret is not None:
-            b64_app_secret = b64encode((app_secret + ":").encode("utf8"))
+        if self.client_secret is not None:
+            b64_client_secret = b64encode((self.client_secret + ":").encode("utf8"))
             authorization = "Basic {secret}".format(
-                secret=b64_app_secret.decode("utf8")
+                secret=b64_client_secret.decode("utf8")
             )
             self.admin_session.headers = {
                 "Authorization": authorization,
                 "X-Nylas-API-Wrapper": "python",
-                "X-Nylas-Client-Id": self.app_id,
+                "X-Nylas-Client-Id": self.client_id,
                 "User-Agent": version_header,
             }
         super(APIClient, self).__init__()
 
+    @property
+    def app_id(self):
+        warn(DeprecationWarning('"app_id" naming convention will be deprecated Sept 2019. '
+                                'Use "client_id" instead.'))
```

Comment:
np: let's say `app_id` will be deprecated in version 5.0.0. Use `client_id` instead.

Reply:
Good idea!